CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+* Version numbers of any other relevant software being used, e.g. operating system
+* A description of what already has been done to solve the problem
+
+The more information that is given up front, the more likely it is that a developer
+will find interest in solving the problem. You will probably get follow-up questions
+after submitting a bug report. Please answer them in a timely manner if you have an
+interest in getting the issue solved.
+
+Finally, please only submit bug reports that are actually related to PROJ.4. If the
+issue materializes in software that uses PROJ.4 it is likely a problem with that
+particular software. Make sure that it actually is a PROJ.4 problem before you submit
+an issue. If you can reproduce the problem only by using tools from PROJ.4 it is
+definitely a problem with PROJ.4.
+
+## Feature requests
+
+Got an idea for a new feature in PROJ.4? Submit a thorough description of the new
+feature in the [issue tracker](https://github.com/OSGeo/proj.4/issues). Please
+include any technical documents that can help the developer make the new feature a
+reality. An example of this could be a publicly available academic paper that
+describes a new projection.
+
+Note that not all feature requests are accepted.
+
+## Write documentation
+
+PROJ.4 is in dire need of better documentation. Any contributiions of documentation
+are greatly appreaciated. The PROJ.4 documentation is available on [proj4.org](ttp://proj4.org).
+The website is generated with [Sphinx](http://www.sphinx-doc.org/en/stable/). Contributions
+to the documentation should be made as [Pull Requests](https://github.com/OSGeo/proj.4/pulls)
+on GitHub.
+
+If you intend to document on of PROJ.4's supported projections please use the
+[Mercator projection](http://proj4.org/projections/merc.html) as a template.
+
+
+## Code contributions
+
+Code contributions can be either bug fixes or new features. The process is the same
+for both, so they will be discussed together in this section.
+
+### Making Changes
+
+* Create a topic branch from where you want to base your work.
+  * You usually should base your topic branch off of the master branch.
+  * To quickly create a topic branch: `git checkout -b my-topic-branch`
+* Make commits of logical units.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure your commit messages are in the [proper format](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+* Make sure you have added the necessary tests for your changes.
+* Make sure that all tests pass
+
+### Submitting Changes
+
+* Push your changes to a topic branch in your fork of the repository.
+* Submit a pull request to the PROJ.4 repository in the OSGeo organization.
+  * If your pull request fixes/references an issue, include that issue number in the pull request. For example:
+
+```
+Wiz the bang
+
+Fixes #123.
+```
+
+* PROJ.4 developers will look at your patch and take an appropriate action.
+
```

Comment:
A more complete walk-through of the steps would probably be useful for newbies. But it may work better as link to a separate document. Anyway - what's here is certainly much better than what wasn't earlier :-)